delivery_state/models/stock_picking.py

```diff
@@ -40,7 +41,34 @@ class StockPicking(models.Model):
     )
 
     def tracking_state_update(self):
-        for picking in self.filtered(lambda p: p.delivery_type):
+        """Call to the service provider API wich should have the method
+        definedin the manner. Can be triggered manually or by cron:
+        <my_provider>_tracking_state_update"""
+        for picking in self.filtered("delivery_type"):
             method = '%s_tracking_state_update' % picking.delivery_type
             if hasattr(picking.carrier_id, method):
                 getattr(picking.carrier_id, method)(picking)
+
+    @api.model
+    def _update_delivery_state(self):
+        """Automated action to query the delivery states to the carriers API.
+        every carrier should implement it 's own method. We split them by
+        delivery type so only those carries with the method update"""
+        pickings = self.search([
+            ("state", "=", "done"),
+            ("delivery_state", "not in",
+                ["customer_delivered", "canceled_shipment"]),
+            # These won't ever autoupdate, so we don't want to evaluate them
+            ("delivery_type", "not in", [False, "fixed", "base_one_rule"]),
```

Comment:
Field is required, so no need to test `False` value:
```suggestion
            ("delivery_type", "not in", ["fixed", "base_one_rule"]),
```

Reply:
It can be `False` if no carrier is defined in the picking (i.e.: internal, receptions)

Reply:
OK, right.